numba/tests/test_dictobject.py

```diff
@@ -1019,3 +1022,189 @@ def test_repr(self):
+        self.assertPreciseEqual(d['velocity'], expect[1])
+        # Ordering is kept
+        for got, exp in zip(d.values(), expect):
+            self.assertPreciseEqual(got, exp)
+
+        # Try deleting
+        self.assertTrue('mass' in d)
+        self.assertTrue('velocity' in d)
+        del d['mass']
+        self.assertFalse('mass' in d)
+        del d['velocity']
+        self.assertFalse('velocity' in d)
+        del expect[0:2]
+
+        for i in range(90):
+            k, v = d.popitem()
+            w = expect.pop()
+            self.assertPreciseEqual(v, w)
+
+        # Trigger a resize
+        expect.append(np.random.random(10))
+        d["last"] = expect[-1]
+
+        # Ordering is kept
+        for got, exp in zip(d.values(), expect):
+            self.assertPreciseEqual(got, exp)
+
+    def test_dict_of_dict_int_keyval(self):
+        def inner_numba_dict():
+            d = TypedDict.empty(
+                key_type=types.intp,
+                value_type=types.intp,
+            )
+            return d
+
+        d = TypedDict.empty(
+            key_type=types.intp,
+            value_type=types.DictType(types.intp, types.intp),
+        )
+
+        def usecase(d, make_inner_dict):
+            for i in range(100):
+                mid = make_inner_dict()
+                for j in range(i + 1):
+                    mid[j] = j * 10000
+                d[i] = mid
+            return d
+
+        got = usecase(d, inner_numba_dict)
+        expect = usecase({}, dict)
+
+        self.assertIsInstance(expect, dict)
+
+        self.assertEqual(dict(got), expect)
+
+        # Delete items
+        for where in [12, 3, 6, 8, 10]:
+            del got[where]
+            del expect[where]
+            self.assertEqual(dict(got), expect)
+
+    def test_dict_of_dict_npm(self):
+        inner_dict_ty = types.DictType(types.intp, types.intp)
+
+        @njit
+        def inner_numba_dict():
+            d = TypedDict.empty(
+                key_type=types.intp,
+                value_type=types.intp,
+            )
+            return d
+
+        @njit
+        def foo(count):
+            d = TypedDict.empty(
+                key_type=types.intp,
+                value_type=inner_dict_ty,
+            )
+            for i in range(count):
+                d[i] = inner_numba_dict()
+                for j in range(i + 1):
+                    d[i][j] = j
+
+            return d
+
+        d = foo(100)
+        ct = 0
+        for k, dd in d.items():
+            ct += 1
+            self.assertEqual(len(dd), k + 1)
+            for kk, vv in dd.items():
+                self.assertEqual(kk, vv)
+
+        self.assertEqual(ct, 100)
+
+    def test_delitem(self):
+        d = TypedDict.empty(types.int64, types.unicode_type)
+        d[1] = 'apple'
+
+        @njit
+        def foo(x, k):
+            del x[1]
+
+        foo(d, 1)
+        self.assertEqual(len(d), 0)
+        self.assertFalse(d)
```

Comment:
Guessing that the ban on reflected types still need writing and testing?

Reply:
right, still need to do this

Reply:
done